libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -291,10 +298,12 @@ method subscribeTopic*(g: GossipSub,
     g.gossipsub[topic].excl(peerId)
 
   libp2p_gossipsub_peers_per_topic_gossipsub
-    .set(g.gossipsub.getOrDefault(topic).len.int64, labelValues = [topic])
+    .set(g.gossipsub[topic].len.int64, labelValues = [topic])
+
+  debug "gossip peers", peers = g.gossipsub[topic].len, topic
 
-  if topic in g.topics:
-    await g.rebalanceMesh(topic)
+  # also rebalance current topic
```

Comment:
Rebalance should only happen on topics we subscribe to, meaning those that are in the `topics` table, this will break things, the previous check is correct.

Reply:
true, for some reason, I imagined this was a `for`, this notation messes up with my eyes!